### PR TITLE
fix rpc error: support escaped strings

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

**Comment by jsdw** [comment]
Ooooh, I see! owned strings needed for when you can't just copy the message straight from the buffer, got it!

**Review by jsdw — APPROVED** [review]

**Review by dvdplm — APPROVED** [review]
Great catch!

File: types/src/v2/error.rs [hunk 1]
```diff
@@ -64,7 +65,7 @@ pub struct ErrorObject<'a> {
 	/// Code
 	pub code: ErrorCode,
 	/// Message
-	pub message: &'a str,
+	pub message: Cow<'a, str>,
```
Comment (maciejhirsz) [inline types/src/v2/error.rs:68]:
```suggestion
	#[serde(borrow)]
	pub message: Cow<'a, str>,
```

Otherwise it always deserializes to owned variant.

Comment (niklasad1) [reply types/src/v2/error.rs:69]: danke :pray:

**Review by maciejhirsz — APPROVED** [review]

**Review by TarikGul — APPROVED** [review]

**Review by insipx — APPROVED** [review]

**Pull request merged**